src/main/kotlin/org/jitsi/rtp/rtcp/CompoundRtcpPacket.kt

```diff
@@ -16,24 +16,49 @@
 
 package org.jitsi.rtp.rtcp
 
-class CompoundRtcpPacket(
-    buffer: ByteArray,
-    offset: Int,
-    length: Int
-) : RtcpPacket(buffer, offset, length) {
-
-    val packets: List<RtcpPacket> = run {
-        var bytesRemaining = length
-        var currOffset = offset
-        val rtcpPackets = mutableListOf<RtcpPacket>()
-        while (bytesRemaining >= RtcpHeader.SIZE_BYTES) {
-            val rtcpPacket = parse(buffer, currOffset)
-            rtcpPackets.add(rtcpPacket)
-            currOffset += rtcpPacket.length
-            bytesRemaining -= rtcpPacket.length
+import org.jitsi.rtp.util.BufferPool
+
+class CompoundRtcpPacket private constructor(
+    holder: Holder
+) : RtcpPacket(holder.buffer, holder.offset, holder.length) {
+
+    val packets: List<RtcpPacket> by lazy { parse(holder) }
+
+    constructor(buffer: ByteArray, offset: Int, length: Int) : this(Holder(buffer, offset, length))
+    constructor(packets: List<RtcpPacket>) : this(createBuffer(packets))
```

Comment:
`vararg` would be slightly more convenient I think (avoids the need for `listOf`)